salt/modules/rabbitmq.py

```diff
@@ -111,15 +117,14 @@ def _get_rabbitmq_plugin():
     This works by taking the rabbitmq-server version and looking for where it
     seems to be hidden in /usr/lib.
     '''
-    rabbitmq = __context__.get('rabbitmq-plugins')
+    global RABBITMQ_PLUGINS
 
-    if rabbitmq is None:
+    if RABBITMQ_PLUGINS is None:
         version = __salt__['pkg.version']('rabbitmq-server').split('-')[0]
-        rabbitmq = ('/usr/lib/rabbitmq/lib/rabbitmq_server-{0}'
-                    '/sbin/rabbitmq-plugins').format(version)
-        __context__['rabbitmq-plugins'] = rabbitmq
+        RABBITMQ_PLUGINS = ('/usr/lib/rabbitmq/lib/rabbitmq_server-{0}'
+                            '/sbin/rabbitmq-plugins').format(version)
```

Comment:
I think we will have to agree to disagree. There are no actual bugs with this current implementation, you are just citing theory and dogma. There is a rationale as to why it is the way it is, which I have already explained. We save what we can when we calculate whether or not RabbitMQ is present, and defer the rest for when we actually need to calculate it, likely because the deferred logic calls into another module (this is my guess, there have been multiple contributors). If calling into another module during `__virtual__` is not frowned upon, then I would say it is cleaner to do it all in `__virtual__`, but I could go either way on this and don't really feel that there is a problem with the current architecture.

Furthermore, this structure was in place prior to this PR, and this is beyond the scope of the intent of this PR. Since this issue is unrelated, it should be its own PR. If you feel strongly that the structure should be changed, you are welcome to submit your own PR to do so. I personally don't feel it is all that important, but would not be offended if you were to change it towards your vision of what it should look like (as long as there isn't any significant performance impact).